src/dom/query.ts

```diff
@@ -86,12 +83,29 @@ function hitTest(node: Element, clientX: number, clientY: number): boolean {
  * ```
  */
 export
-function scrollIfNeeded(area: HTMLElement, elem: HTMLElement, threshold = 0): void {
+function scrollIfNeeded(area: HTMLElement, elem: HTMLElement): void {
   let ar = area.getBoundingClientRect();
   let er = elem.getBoundingClientRect();
-  if (er.top < (ar.top - threshold)) {
-    area.scrollTop -= ar.top - er.top + threshold;
-  } else if (er.bottom > (ar.bottom + threshold)) {
-    area.scrollTop += er.bottom - ar.bottom + threshold;
+  if (er.top < ar.top && er.bottom > ar.bottom) {
+    return;
+  }
+  if (er.top > ar.top && er.bottom < ar.bottom) {
+    return;
+  }
+  if (er.top < ar.top && er.width < ar.width) {
```

Comment:
Okay. I think the basic algorithm of:

* If the element top is below the area bottom, align the tops
* If the element bottom is above the area top, align the bottoms

works okay. It's aggressive about scrolling when it does scroll (i.e., it shows as much of the element as it can).